parcels/kernel.py

```diff
@@ -453,39 +426,26 @@ def get_kernel_compile_files(self):
             dyn_dir = get_cache_dir()
             basename = f"{cache_name}_0"
         lib_path = "lib" + basename
-        src_file_or_files = None
-        if type(basename) in (list, dict, tuple, ndarray):
-            src_file_or_files = [""] * len(basename)
-            for i, src_file in enumerate(basename):
-                src_file_or_files[i] = f"{os.path.join(dyn_dir, src_file)}.c"
-        else:
-            src_file_or_files = f"{os.path.join(dyn_dir, basename)}.c"
```

Comment:
`basename` from `get_kernel_compile_files()` looks to be always a string. I don't know why there are checks for it being an iterable (introduced in c38e927450ff699ba0f4ca08b9c7816bf18cd678)